ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +304,432 @@ def test_ipa_user_s4u2self_pac(self):
         finally:
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
+
+    @pytest.mark.parametrize('netbios_name', ['testrelm', '.TESTRELM',
+                                              'Te8*reALM', 'TEST.REALM'])
+    def test_adtrust_install_with_incorrect_netbios_name(self, netbios_name):
+        """
+        Test that ipa-adtrust-install works with incorrect
+        netbios name
+        """
+        msg = (
+            "ipaserver.install.adtrust: ERROR    \n"
+            "Illegal NetBIOS name [{}].\n\n"
+            "ipaserver.install.adtrust: ERROR    "
+            "Up to 15 characters and only uppercase "
+            "ASCII letters, digits and dashes are allowed."
+            " Empty string is not allowed.\n"
+            "Aborting installation.\n"
+        ).format(netbios_name)
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-a",
+                self.master.config.admin_password,
+                "--netbios-name",
+                netbios_name,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stderr_text
+
+    def test_adtrust_install_with_numerical_netbios_name(self):
+        """
+        Test that ipa-adtrust-install works with numerical
+        netbios name
+        """
+        netbios_name = '1234567'
+        msg = (
+            'NetBIOS domain name will be changed to 1234567'
+        )
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-a",
+                self.master.config.admin_password,
+                "--netbios-name",
+                netbios_name,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stdout_text
+        assert result.returncode == 0
+
+    def test_adtrust_install_with_non_ipa_user(self):
+        """
+        Test that ipa-adtrust-install command works
+        when kinit is done as alias i.e root which is
+        not an ipa user.
+        """
+        msg = (
+            'Unrecognized error during check of admin rights: '
+            'root: user not found'
+        )
+        user = 'root'
+        self.master.run_command(
+            ["kinit", "-E", user, self.master.config.admin_password],
+            raiseonerr=False
```

Comment:
This command is failing in the test:
```
RUN ['kinit', '-E', 'root', 'Secret.123']
Extra arguments (starting with "Secret.123").
...
Exit code: 2
```
The code should be: 
```
self.master.run_command(
            ["kinit", "-E", user], stdin_text=self.master.config.admin_password,
            raiseonerr=False
            )
```
What is the purpose of this test? Is there an associated BZ?

Reply:
@flo-renaud there is no bugzilla associated with this test.
The purpose of the test is to check that since root is an alias to admin, wanted to observe the behavior, where in ipa-adtrust-install should fail when we do kinit as root, since root is not an ipa user.